shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_component_impl.cpp

```diff
@@ -56,12 +70,17 @@ ErrCode TmsClientComponentBaseImpl<Impl>::getName(IString** name)
 {
     OPENDAQ_PARAM_NOT_NULL(name);
 
-    return daqTry([&]
+    StringPtr nameObj;
+    try
     {
-        StringPtr nameObj =this->client->readDisplayName(this->nodeId);
-        *name = nameObj.detach();
-        return OPENDAQ_SUCCESS;
-    });
+        nameObj = this->client->readDisplayName(this->nodeId);
+    }
+    catch(...)
+    {
+        nameObj = "unknown name";
```

Comment:
I'd just return empty strings here, rather than "unknown name", or default to the core behaviour where the local ID is returned, if the name is empty.